sqld/src/lib.rs

```diff
@@ -587,6 +591,54 @@ async fn run_storage_monitor(db_path: PathBuf, stats: Stats) -> anyhow::Result<(
     Ok(())
 }
 
+async fn run_checkpoint_cron(db_path: PathBuf, period: Duration) -> anyhow::Result<()> {
+    const RETRY_INTERVAL: Duration = Duration::from_secs(60);
+    let data_path = db_path.join("data");
+    tracing::info!("setting checkpoint interval to {:?}", period);
+    let mut interval = interval(period);
+    let mut retry = None;
+    loop {
+        if let Some(retry) = retry.take() {
+            sleep(retry).await;
+        } else {
+            interval.tick().await;
+        }
+        retry = match rusqlite::Connection::open(&data_path) {
+            Ok(conn) => unsafe {
+                let start = Instant::now();
+                let mut num_checkpointed: std::ffi::c_int = 0;
+                let rc = rusqlite::ffi::sqlite3_wal_checkpoint_v2(
```

Comment:
this is blocking

Reply:
I've wrapped it with spawn_blocking.

Reply:
@MarinPostma just to double-check -> this isn't running in Tokio main thread, so isn't `block_in_place` more lightweight?